### ISSUE TITLE
[BUG] (C bindings library) Only the first document generated is correct

### ISSUE THREAD

**Issue by anselme16** [wkhtmltopdf/wkhtmltopdf#4449]
**wkhtmltopdf version(s) affected**: 0.12.4 / 0.12.5 / 0.12.6-dev

**OS information**
Windows 7 / Visual C++ 2017

**Description**
The wkhtmltox C library cannot be used twice to generate a document :

- BUG 1 : if `wkhtmltopdf_init` and `wkhtmltopdf_deinit` are called for each document generation, the document is completely corrupted.

- BUG 2 : if the document has a TOC, the second time it is generated, these errors show up during the "Counting pages" and the "Loading TOC" phases, and the resulting pdf has a blank page instead of the TOC : 
```
Error FODC0002 in tag:trolltech.com,2007:QtXmlPatterns:QIODeviceVariable:u: Unknown error
Error XPDY0002 in file:///[PATH_TO_MY_EXE]/test.exe, at line 62, column 0: The focus is undefined.
```

- BUG 3 : to generate a TOC with the C API, this undocumented option must be called on the toc object settings :
`wkhtmltopdf_set_object_setting(tocSettings, "isTableOfContent", "true")`
Please update the C api documentation ( https://wkhtmltopdf.org/libwkhtmltox/pagesettings.html ), i had to dive in your sources for the answer.

**How to reproduce**
These bugs are pretty easy to reproduce, but i can provide a minimal C++ Qt project if necessary.

**Expected behavior**
Multiple document generations should have the same behaviour.

**Possible Solution**  
- Make sure everything is freed and cleaned after a call to `wkhtmltopdf_deinit`
- Update the C API documentation (add the "isTableOfContent" option)

**Comment by PhilterPaper** [comment]
There have been problems reported before with corrupted output when converting multiple documents in one run. Either some internal state was not cleaned up for subsequent runs, or some fixed-name resource was being used by multiple runs in parallel. If you search this ticket list, you might be able to find these older reports.

**Comment by anselme16** [comment]
Mostly Duplicate of #4000

**Comment by ashkulz** [comment]
@anselme16 would appreciate a PR with documentation updates. That said, you shouldn't really call the `deinit` functions unless your application is shutting down.
